airbyte-commons-worker/src/main/java/io/airbyte/workers/helper/PersistConfigHelper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.helper;
+
+import com.google.common.hash.Hashing;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.DestinationApi;
+import io.airbyte.api.client.generated.JobsApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.DestinationIdRequestBody;
+import io.airbyte.api.client.model.generated.DestinationRead;
+import io.airbyte.api.client.model.generated.DestinationUpdate;
+import io.airbyte.api.client.model.generated.JobIdRequestBody;
+import io.airbyte.api.client.model.generated.JobInfoLightRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.api.client.model.generated.SourceUpdate;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.Config;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for workers to persist updates to Source/Destination configs emitted from
+ * AirbyteControlMessages.
+ */
+public class PersistConfigHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PersistConfigHelper.class);
+
+  private final AirbyteApiClient apiClient;
+
+  public PersistConfigHelper(final AirbyteApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  private UUID getConnectionIdFromJobId(final Long jobId) throws ApiException {
```

Comment:
We've seen some failures in the api calls from the data plane to the control plane and are still trying to figure out why.

In the mean time, I've added a [retry wrapper in this PR](https://github.com/airbytehq/airbyte/pull/19693/files#diff-81b9db1a46774bfa5dc46a3800a820b7c6596c799822e2012761fb37d65c9eb5R147). We probably want to use this wrapper for all the api calls in this class.

Reply:
👍🏻 done in [8203432](https://github.com/airbytehq/airbyte/pull/19811/commits/8203432df13dd43adf060dc7704ce68de0da27ef)